docs/source/developer_documentation/reports.md

```diff
@@ -0,0 +1,79 @@
+# Technical Report Documentation
+
+## Creating reports
+### Location of the report code
+The code of the reports is located inside the `rocky` folder.
+
+### Steps to create a new report
+Make sure rocky is running so you can see the changes you are about to make.
+
+1.	Navigate to the `report_types` folder that contains all the different report types.
+2.	Create a new folder with a descriptive name ending with `_report`. Alternatively, you can duplicate an existing folder, modify the names, and clear its contents. (If you’ve done this, continue to step 3.)
+3.	Inside the new folder, create the following files: `__init__.py`, `report.html`, and `report.py`.
+4.	Define a class within `report.py` with the name of your report and include the following variables:
+```
+class YourNameReport(Report):
+    # The id of your report:
+    id = "your-name-report"
+    # The name users will see:
+    name = _("Your Name Report")
+    description = _("Give a description to your new report.")
+    # All the required and optional plugins (can be empty lists):
+    plugins = {"required": ["nmap"], "optional": ["shodan", "nmap-udp", "nmap-ports", "nmap-ip-range"]}
+    # The OOI types that can serve as input to generate this report:
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    # Path of report.html:
+    template_path = "your_name_report/report.html"
+```
+5.	Open `reports/report_types/helpers.py` and add your new class to the `REPORTS` constant list.
+6.	Implement a method within `report.py` to gather the required data for report generation. This data can be used in `report.html`. See the “Collecting data” section for more information.
+7.	Design the HTML structure for your report within `report.html`.
+8.	Save your changes and refresh the page to see the changes you made immediately.
+
+### Collecting data
```

Comment:
We should explain the difference between generate_data and the possibility to optimize using the collect_data method